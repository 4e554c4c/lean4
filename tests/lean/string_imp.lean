#eval ("abc" ++ "cde").length
#eval "abc".popBack
#eval "".popBack
#eval "abcd".popBack
#eval ("abcd".mkIterator.nextn 2).remainingToString
#eval ("abcd".mkIterator.nextn 2).prevToString
#eval ("abcd".mkIterator.nextn 10).remainingToString
#eval ("abcd".mkIterator.nextn 10).prevToString
#eval "foo.Lean".popnBack 5
#eval "foo.Lean".backn 5
#eval "αβγ".popBack
#eval "αβ".length
#eval ("αβcc".mkIterator.next.insert "_foo_").toString
#eval ("αβcc".mkIterator.next.next.insert "_foo_").toString
#eval ("αβcc".mkIterator.next.next.prev.insert "_foo_").toString
#eval ("αβcc".mkIterator.remaining)
#eval ("αβcc".mkIterator.next.remaining)
#eval ("αβcc".mkIterator.next.insert "αbcβ").remaining
#eval (("αβcc".mkIterator.next.insert "αbcβ").remove 2).remaining
#eval (("αβcc".mkIterator.next.insert "αbcβ").remove 2).prev.remaining
#eval ("αβcc".mkIterator.next.toEnd).remaining
#eval "αβcc".mkIterator.offset
#eval "αβcc".mkIterator.next.offset
#eval "αβcc".mkIterator.next.next.offset
#eval ("αβcc".mkIterator.next.setCurr 'a').offset
#eval ("αβcc".mkIterator.next.insert "αbc").offset
#eval ("αβcc".mkIterator.next.insert "αbc").remaining
#eval ("αβcc".mkIterator.insert "αbc").offset
#eval ("αβcc".mkIterator.next.insert "αbcβ").offset
#eval "αβcd".mkIterator.toEnd.offset
#eval "ab\n\nfoo bla".lineColumn 0
#eval "ab\n\nfoo bla".lineColumn 1
#eval "ab\n\nfoo bla".lineColumn 2
#eval "ab\n\nfoo bla".lineColumn 3
#eval "ab\n\nfoo bla".lineColumn 8
#eval "ab\n\nfoo bla".lineColumn 100
