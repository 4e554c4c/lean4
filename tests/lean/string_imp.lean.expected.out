string_imp.lean:1:0: warning: using 'exit' to interrupt Lean
